build-support/bin/generate_github_workflows.py

```diff
@@ -482,137 +504,217 @@ def upload_log_artifacts(self, name: str) -> Step:
         }
 
 
+class RustTesting(Enum):
+    NONE = "NONE"
+    SOME = "SOME"  # Most tests.
+    ALL = "ALL"  # All tests, lint and bench.
+
+
+def bootstrap_jobs(
+    helper: Helper,
+    python_versions: list[str],
+    validate_ci_config: bool,
+    rust_testing: RustTesting,
+) -> Jobs:
+    human_readable_job_name = "Bootstrap Pants"
+
+    if rust_testing == RustTesting.NONE:
+        human_readable_step_name = ""
+        step_cmd = ""
+    elif rust_testing == RustTesting.SOME:
+        human_readable_job_name += ", test Rust"
+        human_readable_step_name = "Test Rust"
+        # We pass --tests to skip doc tests because our generated protos contain
+        # invalid doc tests in their comments. We do not pass --all as BRFS tests don't
+        # pass on GHA MacOS containers.
+        step_cmd = helper.wrap_cmd("./cargo test --tests -- --nocapture")
+    elif rust_testing == RustTesting.ALL:
+        human_readable_job_name += ", test and lint Rust"
+        human_readable_step_name = "Test and lint Rust"
+        # We pass --tests to skip doc tests because our generated protos contain
+        # invalid doc tests in their comments.
+        step_cmd = dedent(
+            """\
+            ./build-support/bin/check_rust_pre_commit.sh
+            ./cargo test --all --tests -- --nocapture
+            ./cargo check --benches
+            """
+        )
+    else:
+        raise ValueError(f"Unrecognized RustTesting value: {rust_testing}")
+
+    if helper.platform in [Platform.LINUX_X86_64]:
+        step_cmd = "sudo apt-get install -y pkg-config fuse libfuse-dev\n" + step_cmd
+    human_readable_job_name += f" ({helper.platform_name()})"
+
+    return {
+        "name": human_readable_job_name,
+        "runs-on": helper.runs_on(),
+        "strategy": {"matrix": {"python-version": python_versions}},
+        "env": DISABLE_REMOTE_CACHE_ENV,
+        "timeout-minutes": 60,
+        "if": IS_PANTS_OWNER,
+        "steps": [
+            *helper.bootstrap_pants(),
+            *(
+                [
+                    {
+                        "name": "Validate CI config",
+                        "run": dedent(
+                            """\
+                    ./pants run build-support/bin/generate_github_workflows.py -- --check
+                    """
+                        ),
+                    }
+                ]
+                if validate_ci_config
+                else []
+            ),
+            *(
+                [
+                    {
+                        "name": human_readable_step_name,
+                        # We pass --tests to skip doc tests because our generated protos contain
+                        # invalid doc tests in their comments.
+                        "run": step_cmd,
+                        "env": {"TMPDIR": f"{gha_expr('runner.temp')}"},
```

Comment:
What was that about in the end?

Reply:
The TMPDIR bit?  We previously only set it on some platforms, but it's the right thing to do in general, so it seemed better to do it uniformly than to maintain the special case for no reason.

Reply:
Ok. It's weird it's needed at all with no doc why.